salt/modules/salt_check.py

```diff
@@ -0,0 +1,449 @@
+#!/usr/bin/env python
+# -*- coding: UTF-8 -*-
+'''This custom salt module makes it easy to test salt states and highstates.
+   Author: William Cannon  william period cannon at gmail dot com'''
+import os
+import os.path
+import yaml
+import salt.client
+import salt.minion
+import salt.config
+
+
+class SaltCheck(object):
+    '''
+    This class implements the salt_check
+    '''
+
+    def __init__(self):
+        self.__opts__ = salt.config.minion_config('/etc/salt/minion')
```

Comment:
I'd rather figure out a way to see if we can get the loader to inject opts in here, but that's a little outside the scope of what you're trying to do, obviously. There might be a way to pull it out of the module namespace here where the loader should be putting it but I'd need to poke around a bit to see for sure.